### PR TITLE
[Refactor] [roleRegistry] Integrates `roleRegistry` admin logic into our existing contracts

### PR DESCRIPTION
**Pull request by jtfirek**
Migrates our contracts to use the `roleRegistry` for admin access control. Each contract defines its own admin role that is then registered with the `roleRegistry` contract.

Obviously some test clean up to be done, but this will be handled upcoming test suite refactor project @vvalecha519 and I will be undertaking.

### CODE REVIEW

File: script/deploys/DeployPhaseOnePointFive.s.sol [hunk 1]
```diff
@@ -126,11 +126,12 @@ contract DeployPhaseOnePointFiveScript is Script {
     }
 
     function setUpAdmins(address _admin) internal {
-        liquidityPool.updateAdmin(_admin, true);
-        regulationsManager.updateAdmin(_admin, true);
-        membershipManager.updateAdmin(_admin, true);
-        membershipNFT.updateAdmin(_admin, true);
-        nftExchange.updateAdmin(_admin);
+        // DEPRECATED
+        // liquidityPool.updateAdmin(_admin, true);
```
Comment (jtfirek) [inline script/deploys/DeployPhaseOnePointFive.s.sol:130]: This refactor broke many of our previous deployment scripts. I just commented them out for now for V and I to handle later in our test refactor project.

File: src/BucketRateLimiter.sol [hunk 2]
```diff
@@ -16,8 +16,8 @@ contract BucketRateLimiter is IRateLimiter, IPausable, Initializable, PausableUp
     BucketLimiter.Limit public limit;
```
Comment (jtfirek) [inline src/BucketRateLimiter.sol:16]: I didn't create an admin role for the `BucketRateLimiter` bc it doesn't expose any admin functionality

File: src/EtherFiAdmin.sol [hunk 3]
```diff
@@ -53,17 +49,24 @@ contract EtherFiAdmin is Initializable, OwnableUpgradeable, UUPSUpgradeable {
...
 
-    mapping(address => bool) public pausers;
+    mapping(address => bool) public DEPRECATED_pausers;
 
     mapping(bytes32 => TaskStatus) public validatorManagementTaskStatus;
     uint16 validatorTaskBatchSize;
 
+    RoleRegistry public roleRegistry;
+
+    bytes32 public constant ETHERFI_ADMIN_ADMIN_ROLE = keccak256("ETHERFI_ADMIN_ADMIN_ROLE");
```
Comment (jtfirek) [inline src/EtherFiAdmin.sol:59]: I tried to keep the name convention consistent, but the resulting name for this contract isn't the best.
Lmk if I should break consistently for a better name

File: src/BucketRateLimiter.sol [hunk 4]
```diff
@@ -43,6 +43,8 @@ contract BucketRateLimiter is IRateLimiter, IPausable, Initializable, PausableUp
     function initializeV2dot5(address _roleRegistry) external onlyOwner {
         require(address(roleRegistry) == address(0x00), "already initialized");
 
+        // TODO: compile list of values in DEPRECATED_pausers to clear out
```
Comment (jtfirek) [inline src/BucketRateLimiter.sol:46]: Good task for V and I's clean up project

File: foundry.toml [hunk 5]
```diff
@@ -4,7 +4,7 @@ out = 'out'
 libs = ['lib']
 fs_permissions = [{ access = "read-write", path = "./release"}, { access = "read", path = "./test" }]
 gas_reports = ["*"]
-optimizer_runs = 2000
+optimizer_runs = 200
```
Comment (jtfirek) [inline foundry.toml:7]: speed up compile time and still avoids stack to deep errors

File: src/EtherFiNodesManager.sol [hunk 6]
```diff
@@ -283,7 +283,7 @@ contract EtherFiNodesManager is
     //  4. wait for 'minWithdrawalDelayBlocks' (= 7 days) delay to be passed
```
Comment (jtfirek) [inline src/EtherFiNodesManager.sol:283]: No changes in this file as it was already migrated in Dave's POC

File: foundry.toml (hunk 5 above)
Comment (solipsis) [reply foundry.toml:7]: 2000 was already a concession. You should think of this  number as "how many times do I expect this contract to be called over its lifetime". If that number 10k+ then setting this to 200 will most likely substantially increase gas usage in the long run. We should just make sure we run gas comparisons and consider the tradeoffs before changing this number

File: src/AuctionManager.sol [hunk 7]
```diff
@@ -46,10 +46,16 @@ contract AuctionManager is
...
-    mapping(address => bool) public admins;
+    mapping(address => bool) public DEPRECATED_admins;
 
     RoleRegistry public roleRegistry;
 
+    //--------------------------------------------------------------------------------------
+    //-------------------------------------  ROLES  ---------------------------------------
+    //--------------------------------------------------------------------------------------
+
+    bytes32 public constant AUCTION_ADMIN_ROLE = keccak256("AUCTIN_ADMIN_ROLE");
```
Comment (solipsis) [inline src/AuctionManager.sol:57]: Typo :)

File: src/NodeOperatorManager.sol [hunk 8]
```diff
@@ -36,11 +36,17 @@ contract NodeOperatorManager is INodeOperatorManager, IPausable, Initializable,
...
+    mapping(address => bool) public DEPRECATED_admins;
     mapping(address => mapping(ILiquidityPool.SourceOfFunds => bool)) public operatorApprovedTags;
 
     RoleRegistry public roleRegistry;
 
+    //--------------------------------------------------------------------------------------
+    //-------------------------------------  ROLES  ---------------------------------------
+    //--------------------------------------------------------------------------------------
+
+    bytes32 public constant NODE_OPERATOR_ADMIN_ROLE = keccak256("NODE_OPERATOR_ADMIN_ROLE");
```
Comment (solipsis) [inline src/NodeOperatorManager.sol:48]: Purely for consistency lets have `NODE_OPERATOR_MANAGER_ADMIN_ROLE`

File: test/TestSetup.sol [hunk 9]
```diff
@@ -411,6 +407,11 @@ contract TestSetup is Test {
         bucketRateLimiter.setCapacity(40 ether);
         bucketRateLimiter.setRefillRatePerSecond(1 ether);
 
+        if (forkEnum == MAINNET_FORK || forkEnum == TESTNET_FORK) {
```
Comment (solipsis) [inline test/TestSetup.sol:410]: what's the reason to limit this. Wouldn't we always want to upgrade to latest version, even if on say a tenderly fork?

**Review by solipsis — APPROVED** [review]
Looks good. A few small nits to clean up. We need to fully redo our roles for testing setup during audit downtime. It is really confusing that alice is owner and admin half the time

File: src/AuctionManager.sol (hunk 7 above)
Comment (jtfirek) [reply src/AuctionManager.sol:57]: fixed

File: src/EtherFiOracle.sol [hunk 10]
```diff
@@ -344,9 +353,4 @@ contract EtherFiOracle is Initializable, OwnableUpgradeable, PausableUpgradeable
     }
 
     function _authorizeUpgrade(address newImplementation) internal override onlyOwner {}
-
-    modifier isAdmin() {
-        require(admins[msg.sender] || msg.sender == owner(), "EtherFiAdmin: not an admin");
-        _;
-    }
 }
```
[the remark this answers is not in the document]
Comment (jtfirek) [reply src/EtherFiOracle.sol:356]: fixed

File: src/NodeOperatorManager.sol (hunk 8 above)
Comment (jtfirek) [reply src/NodeOperatorManager.sol]: fixed

File: test/TestSetup.sol (hunk 9 above)
Comment (jtfirek) [reply test/TestSetup.sol:410]: I was also unsure of the purpose, but I was just keep this existing logic intact:
https://github.com/etherfi-protocol/smart-contracts/blob/2cebfd7e87cca237d4444b4632d22aff439ce132/test/TestSetup.sol#L398-L404

Noted as something to look into for the test refactor

**Comment by jtfirek** [comment]
> Looks good. A few small nits to clean up. We need to fully redo our roles for testing setup during audit downtime. It is really confusing that alice is owner and admin half the time

Agreed! this task made me appreciate how badly a refactor of the test suite is needed

**Pull request merged**